### PR TITLE
introduces ranker.clj to move ranker functions from the sheduler.clj file

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- introduces ranker.clj

## Why are we making these changes?

Makes it easier to maintain (find and edit) ranker functions. Shortens the `scheduler.clj` file too in the process.

### CODE REVIEW

File: scheduler/src/cook/mesos/scheduler.clj [hunk 1]
```diff
@@ -1278,192 +1277,6 @@
     {:error-handler (fn [e]
                       (log/error e "Failed to kill cancelled tasks!"))}))
 
-(defn get-user->used-resources
```
Comment (shamsimam) [inline scheduler/src/cook/mesos/scheduler.clj]: This function was deleted as it is not in use.

File: scheduler/src/cook/mesos/ranker.clj [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            [cook.mesos.share :as share]
+            [cook.mesos.util :as util]
+            [datomic.api :as d]
+            [metrics.meters :as meters]
+            [metrics.timers :as timers]
+            [plumbing.core :as pc]))
+
+(timers/deftimer [cook-mesos scheduler filter-offensive-jobs-duration])
+
+(defn is-offensive?
```
Comment (shamsimam) [inline scheduler/src/cook/mesos/ranker.clj]: The offensive filter related functions have been moved above the ranking related functions.

**Comment by dposada** [comment]
@shamsimam Would you be opposed to merging this after getting the cook pools PR merged?

**Comment by shamsimam** [comment]
@dposada nope, we can merge this later.

**Comment by dposada** [comment]
Can we delete the branch?

**Pull request closed without merging**